src/openforms/submissions/models/submission.py

```diff
@@ -656,3 +663,28 @@ def get_prefilled_data(self):
                 variable.key: variable.value for variable in prefill_vars
             }
         return self._prefilled_data
+
+    @lru_cache
+    def get_cosigner_email(self) -> str | None:
```

Comment:
I believe `lru_cache` is not thread-safe and as we deploy uwsgi with multiple threads, this cache would be shared by different requests (each request gets its own thread) in the same process. In this case, it's mitigated because it's on an Submission instance which is typically created per thread, but I think I'd rather see [`django.utils.functional.cached_property` ](https://docs.djangoproject.com/en/3.2/ref/utils/#django.utils.functional.cached_property)for this.

Reply:
The documentation says it is threadsafe: https://docs.python.org/3/library/functools.html#functools.lru_cache.
Should I still change it to a [django.utils.functional.cached_property](https://docs.djangoproject.com/en/3.2/ref/utils/#django.utils.functional.cached_property)?

Reply:
I think it's thread safe in the sense that multiple threads can use the cache, but the contents of the cache is shared across threads, but I'm not 100% sure about this!